src/main/java/com/openlattice/shuttle/dates/TimeZones.java

```diff
@@ -34,4 +41,24 @@
     private TimeZones() {
     }
 
+    private static final Logger logger = LoggerFactory.getLogger( Parsers.class );
+
+    public static Optional<TimeZone> checkTimezone( Optional<String> timezoneString ) {
+        if ( timezoneString.isPresent() ) {
+            String timezoneId = timezoneString.get();
+
+            Optional<TimeZone> timezone = Optional.of(TimeZone.getTimeZone( timezoneId ));
+
+            if ( timezone.orElse( Constants.DEFAULT_TIMEZONE ).getID().equals( timezoneId ) ) {
+                throw new IllegalArgumentException(
+                        "Invalid timezone id " + timezoneId + " requested" );
+            }
+
+            return timezone;
+
+        } else {
```

Comment:
super minor, but you don't need the else statement here -- you can just `return Optional.empty();`